File: Ywy/JavaScript/level2/Nara1234.js
export default class Nara1234 {
    solution = (n) => {
        const answer = [];
        let digit = 0;
        while (n > 0) {
            digit = n % 3;
            if (digit === 0) {
                n--;
                digit = 4;
            }
            n = (n / 3) << 0;
            answer.unshift(digit);
        }
        return answer.join("");
    };
}

File: Ywy/JAVA/src/level2/Nara1234.java
package level2;

public class Nara1234 {
	
	public String solution(int n) {
		StringBuffer sb = new StringBuffer();
		int digit = 0;
		while(n > 0) {
			digit = n % 3;
			if (digit == 0) {
				n--;
				digit = 4;
			}
			n /= 3;
			sb.append(digit);
		}
		return sb.reverse().toString();
	}
}
